addons/website_slides/models/slide_channel.py

```diff
@@ -342,24 +428,45 @@ def _get_default_enroll_msg(self):
     def _compute_enroll(self):
         self.filtered(lambda channel: channel.visibility == 'members').enroll = 'invite'
 
+    @api.depends('channel_partner_all_ids', 'channel_partner_all_ids.member_status')
+    def _compute_partner_ids(self):
+        for record in self:
+            enrolled_partner_ids = record.channel_partner_ids.partner_id.ids or []
+            record.partner_ids = record.partner_all_ids.filtered(lambda partner: partner.id in enrolled_partner_ids)
+
+    def _search_partner_ids(self, operator, value):
+        if isinstance(value, int) and operator == 'in':
+            value = [value]
```

Comment:
So the answer is basically to make sure search works with `('....partner_ids', 'in', user.partner_id.id)` shaped leafs, in the forum security rules. Indeed, (after a bit of search) they can be used in domains and changed to [user.partner_id.id] inside the main "domain to query" method in expression.py in odoo/osv (see **parse** method).

However, in this case, when searching here on partner_id, we search on the partner_id field ,i.e. on a Many2one on the `slide.channel.partner` model, and when searching, an error is thrown, **because the code will not transform the right leaf term to a list, and will throw an error in __leaf_to_sql method for the 'in' operator.** For many2many for instance, the non-iterable 'right' are cast to a list in the query method, so it works in practice, and such domains can be found in numerous places in the codebase.

I hope this is clear ^^ I tried to summarize what I found but that code is quite low / long. We may also update (noupdate is set though -> upg script needed) the rules in forum to **partner_id.idS** but there may be other occurrences of `'partner_ids','in',int` leaves (even later on). I guess we prefer to be able to support that case?

Reply:
Thanks for indulging me. 
It makes sense, and I guess it's simple enough to add it here, but it seems like something that could be supported at framework level, unless there is a reason not to do it generally...? (It may well be an issue in our security rules, that said, how do they work now as you add this search method?)